scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,593 @@
+                        sample_kwargs["return_sites"] is not None
+                    ):
+                        return_sites = np.array(sample_kwargs["return_sites"])
+                        return_sites = return_sites[
+                            np.isin(
+                                return_sites,
+                                list(
+                                    self.module.model.list_obs_plate_vars()[
+                                        "sites"
+                                    ].keys()
+                                ),
+                            )
+                        ]
+                        if len(return_sites) == 0:
+                            sample_kwargs_obs_plate["return_sites"] = [return_sites]
+                        else:
+                            sample_kwargs_obs_plate["return_sites"] = list(return_sites)
+                    else:
+                        sample_kwargs_obs_plate["return_sites"] = list(
+                            self.module.model.list_obs_plate_vars()["sites"].keys()
+                        )
+                    sample_kwargs_obs_plate["show_progress"] = False
+                    samples = self._get_posterior_samples(
+                        args, kwargs, **sample_kwargs_obs_plate
+                    )
+
+                    # find plate dimension
+                    trace = poutine.trace(self.module.model).get_trace(*args, **kwargs)
+                    obs_plate = {
+                        name: site["cond_indep_stack"][0].dim
+                        for name, site in trace.nodes.items()
+                        if site["type"] == "sample"
+                        if any(
+                            f.name == self.module.model.list_obs_plate_vars()["name"]
+                            for f in site["cond_indep_stack"]
+                        )
+                    }
+
+                else:
+                    args, kwargs = self.module._get_fn_args_from_batch(tensor_dict)
+                    args = [a.to(device) for a in args]
+                    kwargs = {k: v.to(device) for k, v in kwargs.items()}
+                    self.to_device(device)
+
+                    samples_ = self._get_posterior_samples(
+                        args, kwargs, **sample_kwargs_obs_plate
+                    )
+                    samples = {
+                        k: np.array(
+                            [
+                                np.concatenate(
+                                    [samples[k][i], samples_[k][i]],
+                                    axis=list(obs_plate.values())[0],
+                                )
+                                for i in range(len(samples[k]))
+                            ]
+                        )
+                        for k in samples.keys()
+                    }
+                i += 1
+
+        # sample global parameters
+        i = 0
+        for tensor_dict in train_dl:
```

Comment:
`next(iter(train_dl))`

Reply:
what do you mean?

Reply:
I think you can use that to get the output of the data loader without having to have this for loop and then the break statement.

Reply:
`tensor_dict  = next(iter(train_dl))` like this?